### ISSUE TITLE
[Bug]: Warning: Undefined array key "context"

### ISSUE THREAD

**Issue by skoch-twocream** [pimcore/pimcore#9847]
### Expected behavior

No error ;-)

### Actual behavior

![image](https://user-images.githubusercontent.com/54348050/126775960-3c325716-070e-47d7-a5ff-a15e4b0856e0.png)

### Steps to reproduce

The error occurs on the classification store by using a select field with an option provier inside the grid view.

**Comment by skoch-twocream** [comment]
Wow... github lost the marked line?

I fix this when I am at home...

```diff
    public function getDataForGrid($data, $object = null, $params = [])
    {
        $optionsProvider = DataObject\ClassDefinition\Helper\OptionsProviderResolver::resolveProvider(
            $this->getOptionsProviderClass(),
            DataObject\ClassDefinition\Helper\OptionsProviderResolver::MODE_SELECT
        );

        if ($optionsProvider) {
-            $context = $params['context'] ? $params['context'] : [];
+            $context = $params['context'] ?? [];
            $context['object'] = $object;
            if ($object) {
                $context['class'] = $object->getClass();
            }

            $context['fieldname'] = $this->getName();
            $options = $optionsProvider->{'getOptions'}($context, $this);
            $this->setOptions($options);

-            if ($params['purpose'] == 'editmode') {
+            if (isset($params['purpose']) && $params['purpose'] == 'editmode') {
                $result = $data;
            } else {
                $result = ['value' => $data ?? null, 'options' => $this->getOptions()];
            }

            return $result;
        }

        return $data;
    }
```

File: https://github.com/pimcore/pimcore/blob/10.x/models/DataObject/ClassDefinition/Data/Select.php#L499

**Comment by dvesh3** [comment]
Fixed by #9868
